src/yaml/schema/core.cr

```diff
@@ -0,0 +1,168 @@
+# Provides utility methods for the YAML core schema,
+# as specified in http://www.yaml.org/spec/1.2/spec.html#id2804923,
+# with some additional types from http://www.yaml.org/type/:
+# - bool
+# - int
+# - float
+# - timestamp
+module YAML::Schema::Core
+  # Deserializes a YAML document.
+  #
+  # Same as `YAML.parse`.
+  def self.parse(data : String | IO)
+    Parser.new data, &.parse
+  end
+
+  # Deserializes multiple YAML documents.
+  #
+  # Same as `YAML.parse_all`.
+  def self.parse_all(data : String | IO)
+    Parser.new data, &.parse_all
+  end
+
+  # Assuming the *pull_parser* is positioned in a scalar,
+  # parses it according to the core schema, taking the
+  # scalar's stlye and tag into account.
+  def self.parse_scalar(pull_parser : YAML::PullParser) : Type
+    string = pull_parser.value
+
+    # Non-plain scalar is always a string
+    unless pull_parser.scalar_style.plain?
+      return string
+    end
+
+    # Check for core schema tags
+    case pull_parser.tag
+    when "tag:yaml.org,2002:str"
+      return string
+    when "tag:yaml.org,2002:float"
+      return string.to_f64
+    end
+
+    parse_scalar(string)
+  end
+
+  # Parses a string according to the core schema, assuming
+  # the string had a plain style.
+  #
+  # ```
+  # YAML::Schema::Core.parse_scalar("hello") # => "hello"
+  # YAML::Schema::Core.parse_scalar("1.2")   # => 1.2
+  # YAML::Schema::Core.parse_scalar("false") # => false
+  # ```
+  def self.parse_scalar(string : String) : Type
+    case string
+    when .empty?
```

Comment:
Why not put this with the other null checks?

`when "~", "null", "Null", "NULL", .empty?`

Reply:
My guess is to make it more readable.

Reply:
Oh, I totally forgot I could write it like that. I'll change it :-)